spec/lib/site_settings/deprecated_settings_spec.rb

```diff
@@ -0,0 +1,223 @@
+      SiteSetting.load_settings(deprecated_test)
+    end
+
+    it "should not act as a proxy to the new methods" do
+      deprecate_override!(["use_https", "force_https", override, "0.0.1"])
+
+      SiteSetting.use_https = false
+
+      expect(SiteSetting.force_https).to eq(true)
+      expect(SiteSetting.force_https?).to eq(true)
+    end
+
+    it "should log warnings when deprecated settings are called" do
+      deprecate_override!(["use_https", "force_https", override, "0.0.1"])
+
+      logger =
+        track_log_messages do
+          expect(SiteSetting.use_https).to eq(true)
+          expect(SiteSetting.use_https?).to eq(true)
+        end
+      expect(logger.warnings.count).to eq(3)
+
+      logger = track_log_messages { SiteSetting.use_https(warn: false) }
+      expect(logger.warnings.count).to eq(0)
+    end
+  end
+
+  describe "when overriding deprecated settings" do
+    let(:override) { true }
+    let(:deprecated_test) { "#{Rails.root}/spec/fixtures/site_settings/deprecated_test.yml" }
+
+    before do
+      SiteSetting.force_https = true
+      SiteSetting.load_settings(deprecated_test)
+    end
+
+    it "should act as a proxy to the new methods" do
+      deprecate_override!(["use_https", "force_https", override, "0.0.1"])
+
+      SiteSetting.use_https = false
+
+      expect(SiteSetting.force_https).to eq(false)
+      expect(SiteSetting.force_https?).to eq(false)
+    end
+
+    it "should log warnings when deprecated settings are called" do
+      deprecate_override!(["use_https", "force_https", override, "0.0.1"])
+
+      logger =
+        track_log_messages do
+          expect(SiteSetting.use_https).to eq(true)
+          expect(SiteSetting.use_https?).to eq(true)
+        end
+      expect(logger.warnings.count).to eq(2)
+
+      logger = track_log_messages { SiteSetting.use_https(warn: false) }
+      expect(logger.warnings.count).to eq(0)
+    end
+  end
+
+  describe "when overriding a trust level setting with a group setting" do
+    let(:override) { false }
+    let(:deprecated_test) { "#{Rails.root}/spec/fixtures/site_settings/deprecated_test.yml" }
+
+    before { SiteSetting.load_settings(deprecated_test) }
+
+    context "when getting an old TrustLevelSetting" do
+      before do
+        deprecate_override!(
+          ["min_trust_level_to_allow_invite", "invite_allowed_groups", override, "0.0.1"],
+        )
+      end
+
+      it "uses the minimum trust level from the trust level auto groups in the new group setting" do
+        SiteSetting.invite_allowed_groups =
+          "#{Group::AUTO_GROUPS[:trust_level_3]}|#{Group::AUTO_GROUPS[:trust_level_4]}"
+        expect(SiteSetting.min_trust_level_to_allow_invite).to eq(TrustLevel[3])
+      end
+
+      it "returns nil if there are no trust level auto groups in the new group setting" do
+        SiteSetting.invite_allowed_groups = Fabricate(:group).id.to_s
+        expect(SiteSetting.min_trust_level_to_allow_invite).to eq(nil)
```

Comment:
I think this should return the admin trust level group instead to be safe. Otherwise, there is a chance that people might be doing `user.trust_level >= SiteSetting.min_trust_level_to_allow_invite.to_i` and we can potentially end up with a security hole.

Reply:
Yep fair enough

Reply:
Ah wait though, in the case of `TrustLevelSetting` this is not possible...best we can do is TL4